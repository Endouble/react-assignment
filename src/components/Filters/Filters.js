import React from 'react';

const Filters = (props) => {
    return <React.Fragment>
      <p> filters </p>
    </React.Fragment>
};

export default Filters;