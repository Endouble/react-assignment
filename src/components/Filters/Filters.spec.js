import React from 'react';
import { shallow, mount } from 'enzyme';

import Filters from './Filters';

describe('Filters Test', () => {
  let shallowFilters;

  const props = {
    filterCallback: jest.fn()
  };

  const getShallowFilter = () => {
    if (!shallowFilters) {
      shallowFilters = shallow(<Filters {...props} />);
    }
    return shallowFilters;
  };

  beforeEach(() => {
    shallowFilters = undefined;
  });

  test('Component should render without crashing', () => {
    expect(getShallowFilter()).toBeDefined();
  });

  test('Component should render correctly', () => {
    expect(getShallowFilter()).toMatchSnapshot();
  });

  test('Component should call filterCallback on click over submit', () => {
    const filter = mount(<Filters {...props} />);
    const filterForm = filter.find('Form');
    filterForm.simulate('submit', { preventDefault: jest.fn(), key: 'Enter' });
    expect(props.filterCallback).toHaveBeenCalled();
  });
});