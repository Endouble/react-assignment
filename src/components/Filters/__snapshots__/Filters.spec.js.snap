// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Test Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    filterCallback={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": "form",
      "children": Array [
        <FormField>
          <label>
                    Search by character's name
          </label>
          <input
                    id="name"
                    placeholder="Eg. R2D2"
          />
</FormField>,
        <Button
          as="button"
          primary={true}
          type="submit"
>
          Search
</Button>,
      ],
      "onSubmit": [Function],
      "role": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <label>
              Search by character's name
</label>,
            <input
              id="name"
              placeholder="Eg. R2D2"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search by character's name",
            },
            "ref": null,
            "rendered": "Search by character's name",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "name",
              "placeholder": "Eg. R2D2",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": null,
            "type": "input",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": "Search",
          "primary": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "form",
        "children": Array [
          <FormField>
            <label>
                        Search by character's name
            </label>
            <input
                        id="name"
                        placeholder="Eg. R2D2"
            />
</FormField>,
          <Button
            as="button"
            primary={true}
            type="submit"
>
            Search
</Button>,
        ],
        "onSubmit": [Function],
        "role": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label>
                Search by character's name
</label>,
              <input
                id="name"
                placeholder="Eg. R2D2"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search by character's name",
              },
              "ref": null,
              "rendered": "Search by character's name",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "name",
                "placeholder": "Eg. R2D2",
              },
              "ref": Object {
                "current": null,
              },
              "rendered": null,
              "type": "input",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "Search",
            "primary": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
